lib/matplotlib/axes/_subplots.py

```diff
@@ -173,7 +173,8 @@ def _make_twin_axes(self, *kl, **kwargs):
         """
         from matplotlib.projections import process_projection_requirements
         if 'sharex' in kwargs and 'sharey' in kwargs:
-            raise ValueError("Twinned Axes may share only one axis.")
+            if kwargs["sharex"] is not self and kwargs["sharey"] is not self:
```

Comment:
Please add a comment above this line:
```
# The following line is added in v2.2 to avoid breaking Seaborn,
# which currently uses this internal API.